core/server/master/src/main/java/alluxio/master/metastore/heap/HeapInodeStore.java

```diff
@@ -78,6 +81,25 @@ public void remove(Long inodeId) {
     mInodes.remove(inodeId);
   }
 
+  @Override
+  public void writeNewInode(MutableInode<?> inode) {
+    mInodes.compute(inode.getId(), (k, existingInode) -> {
+      if (existingInode != null && !existingInode.getName().equals(inode.getName())) {
+        LOG.error(
+            "[InodeTreeCorruption] writing inode {} with id {}, parent id {} "
+                + "but a different inode name {} parent id {} already exists. "
+                + "Your journal files are probably corrupted!",
+            inode.getName(), inode.getId(), inode.getParentId(),
+            existingInode.getName(), existingInode.getParentId());
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("[InodeTreeCorruption] Existing inode: {}, new written inode: {}",
```

Comment:
i think we can just log the full path in the error msg so no need for the extra debug one

Reply:
traversing and logging all inode on the full path in the format of protobuf is time consuming and we only want to do it when the debug logging is turned on